pkg/arvo/app/chat-cli.hoon

```diff
@@ -17,8 +17,15 @@
 ::
 |%
 +$  card  card:agent:gall
-+$  state
-  $:  grams=(list mail)                             ::  all messages
+::
++$  versioned-state
+  $%  state-1
+      state-0
+  ==
+::
++$  state-1
+  $:  %1
+      grams=(list mail)                             ::  all messages
```

Comment:
Why are we storing grams when we could scry them on-demand from chat-store? Seems unnecessary.

Reply:
idk how you would solve for message selection/expansion in that case though. This `(list mail)` contains messages from all chats. They get rendered in the same stream, and numbered sequentially.

`;;;` gets you the third-most-recent message, as plucked from this. I don't think you can make such queries against chat-store. Maybe you could "get all messages from everything, merge-sorted-by-timestamp", but performance aside I doubt the ordering that gives you is guaranteed to be identical to the `%fact`/rendering order.

Reply:
Alternatively you could make the `chat-cli` "get message interface" take both a channel and a message number, and then you wouldn't have any real problem with either rendering order, which would be guaranteed, or performance, which would be a single scry, and quite reasonable.

Reply:
Giving separate numeric indicators for messages from different channels seems potentially confusing when we're rendering them in the same stream. I don't disagree that the current implementation is a bit of a hack, but consider fixing that out of scope for now.

Reply:
Okay. I feel that this should be prioritized after the release. @vvisigoth Storing *double* chat's state seems entirely unnecessary and like it will impede our ability to properly export/import chat data down the line.